lib/src/api/opt/mod.rs

```diff
@@ -395,3 +416,403 @@ where
+
+			let simple_json = into_json(value.clone(), true);
+			assert_eq!(
+				simple_json,
+				json!({ "type": "LineString", "coordinates": [[0., 0.], [10., 0.]]})
+			);
+
+			let json = into_json(value.clone(), false);
+			assert_eq!(json, json!(line_string));
+
+			let response: LineString = from_value(value).unwrap();
+			assert_eq!(response, line_string);
+		}
+
+		#[test]
+		fn geometry_polygon() {
+			let polygon = polygon![
+				(x: -111., y: 45.),
+				(x: -111., y: 41.),
+				(x: -104., y: 41.),
+				(x: -104., y: 45.),
+			];
+			let value = Value::Geometry(sql::Geometry::Polygon(polygon.clone()));
+
+			let simple_json = into_json(value.clone(), true);
+			assert_eq!(
+				simple_json,
+				json!({ "type": "Polygon", "coordinates": [[
+					[-111., 45.],
+					[-111., 41.],
+					[-104., 41.],
+					[-104., 45.],
+					// Looks like we have to append the first pair. Is this expected @tobiemh?
+					[-111., 45.],
+				]]})
+			);
+
+			let json = into_json(value.clone(), false);
+			assert_eq!(json, json!(polygon));
+
+			let response: Polygon = from_value(value).unwrap();
+			assert_eq!(response, polygon);
+		}
+
+		#[test]
+		fn geometry_multi_point() {
+			let multi_point: MultiPoint =
+				vec![point! { x: 0., y: 0. }, point! { x: 1., y: 2. }].into();
+			let value = Value::Geometry(sql::Geometry::MultiPoint(multi_point.clone()));
+
+			let simple_json = into_json(value.clone(), true);
+			assert_eq!(
+				simple_json,
+				json!({ "type": "MultiPoint", "coordinates": [[0., 0.], [1., 2.]]})
+			);
+
+			let json = into_json(value.clone(), false);
+			assert_eq!(json, json!(multi_point));
+
+			let response: MultiPoint = from_value(value).unwrap();
+			assert_eq!(response, multi_point);
+		}
+
+		#[test]
+		fn geometry_multi_line() {
+			let multi_line = MultiLineString::new(vec![line_string![
+					( x: 0., y: 0. ),
+					( x: 1., y: 2. ),
+			]]);
+			let value = Value::Geometry(sql::Geometry::MultiLine(multi_line.clone()));
+
+			let simple_json = into_json(value.clone(), true);
+			assert_eq!(
+				simple_json,
+				json!({ "type": "MultiLineString", "coordinates": [[[0., 0.], [1., 2.]]]})
+			);
+
+			let json = into_json(value.clone(), false);
+			assert_eq!(json, json!(multi_line));
+
+			let response: MultiLineString = from_value(value).unwrap();
+			assert_eq!(response, multi_line);
+		}
+
+		#[test]
+		fn geometry_multi_polygon() {
+			let multi_polygon: MultiPolygon = vec![polygon![
+				(x: -111., y: 45.),
+				(x: -111., y: 41.),
+				(x: -104., y: 41.),
+				(x: -104., y: 45.),
+			]]
+			.into();
+			let value = Value::Geometry(sql::Geometry::MultiPolygon(multi_polygon.clone()));
+
+			let simple_json = into_json(value.clone(), true);
+			assert_eq!(
+				simple_json,
+				json!({ "type": "MultiPolygon", "coordinates": [[[
+					[-111., 45.],
+					[-111., 41.],
+					[-104., 41.],
+					[-104., 45.],
+					// Looks like we have to append the first pair. Is this expected @tobiemh?
```

Comment:
@tobiemh 👆

Reply:
Hey @rushmorem could you clarify what you mean here? Is something happening which is unexpected? The polygon has to close (the last coordinate needs to equal the first coordinate, but I'm not sure that was what your question was...

Reply:
Ah I see! That makes sense, I didn't think of that!